openeo/rest/datacube.py

```diff
@@ -187,9 +187,14 @@ def load_collection(
                     "Unexpected parameterized `spatial_extent` in `load_collection`:"
                     f" expected schema with type 'object' but got {spatial_extent.schema!r}."
                 )
+        valid_geojson_types = [
+            "Polygon", "MultiPolygon", "GeometryCollection", "FeatureCollection"
```

Comment:
```suggestion
            "Polygon", "MultiPolygon", "Feature", "FeatureCollection"
```

Usage of GeometryCollections is discouraged, so I would not add it here in this new feature.
Additionally, type "Feature" should be allowed

Reply:
Another thing: if you move `valid_geojson_types` just before the `_get_geometry_argument` call, you can make the next `if` an `elif`